app/abilities/ability.rb

```diff
@@ -60,7 +60,10 @@ def initialize(user)
           can [:index, :show, :resolve, :ignore, :reopen], Issue
           can :create, IssueComment
           can [:new, :create, :edit, :update, :destroy], Redaction
-          can [:new, :edit, :create, :update, :revoke, :revoke_all], UserBlock
+          can [:new, :create, :revoke, :revoke_all], UserBlock
+          can :update, UserBlock, :creator => user
+          can :update, UserBlock, :revoker => user
+          can :update, UserBlock, :active? => true
```

Comment:
Does this act as an AND or an OR of the conditions? Because neither seems right? Surely we want active AND (creator OR revoker) as the test?

Reply:
It's [OR](https://github.com/CanCanCommunity/cancancan/blob/develop/docs/combine_abilities.md) and that's what we want. Before this PR you had to be either a creator or a revoker to edit. Now additionally you can become a revoker by editing. If the block is active, it has no revoker yet. Previously you had to first revoke it, that made you a revoker, then you could edit its message. Now you can edit its message and revoke it at the same time.

Reply:
Thanks for the explanation. I had missed that this was inside a moderator test and so only applied to moderators...